### PR TITLE
[wasm][debugger] Show pointer type+address

### PR DESCRIPTION
**Pull request by radical**

### CODE REVIEW

**Comment by radical** [comment]
This is based on https://github.com/mono/mono/pull/19535, so it includes changes from that. Once it gets merged, I can rebase this.

**Comment by lewing** [comment]
@monojenkins build failed

File: sdks/wasm/src/library_mono.js [hunk 1]
```diff
@@ -441,31 +484,53 @@ var MonoSupportLib = {
...
+		case "getter":
+			MONO._mono_wasm_add_getter_var (str_value);
+			break;
+
+		case "array":
+			MONO._mono_wasm_add_array_var (str_value, int_value);
+			break;
+
+		default:
+			throw new Error (`Unknown type for value: ${type}`);
```
Comment (lewing) [inline sdks/wasm/src/library_mono.js:543]: an error here will probably surface in a really bad way maybe just log it and set a string to the value?

File: mono/mini/mini-wasm-debugger.c [hunk 2]
```diff
@@ -765,12 +759,18 @@ read_enum_value (const char *mem, int type)
 	return 0;
 }
 
+static void
+mono_wasm_add_number_var (guint64 val)
```
Comment (lewing) [inline mono/mini/mini-wasm-debugger.c:763]: just call mono_wasm_add_typed_value ("number", NULL,  everywhere

File: sdks/wasm/src/library_mono.js [hunk 3]
```diff
@@ -433,39 +433,129 @@ var MonoSupportLib = {
...
+					description: symbol_str
+				}
+			});
 			}
-		});
-	},
+			break;
+
+		default: {
+			console.log (`Error: mono_wasm_add_typed_value: Unknown type for value: ${type}`);
```
Comment (lewing) [inline sdks/wasm/src/library_mono.js:543]:
```suggestion
			console.log (`Error: mono_wasm_add_typed_value: Unknown type for value: ${type_str}`);
```

File: sdks/wasm/src/library_mono.js [hunk 4]
```diff
@@ -433,39 +433,129 @@ var MonoSupportLib = {
...
 
-	mono_wasm_add_bool_var: function(var_value) {
-		MONO.var_info.push({
-			value: {
-				type: "boolean",
-				value: var_value != 0,
-			}
-		});
-	},
+	mono_wasm_add_typed_value: function (type, str_value, int_value) {
```
Comment (lewing) [inline sdks/wasm/src/library_mono.js:487]: int_value is misleading, value would be better

**Comment by lewing** [comment]
@monojenkins build failed

**Comment by lewing** [comment]
@monojenkins build Linux AArch64 Coop Suspend

**Comment by lewing** [comment]
@monojenkins build failed

**Comment by lewing** [comment]
@monojenkins build Linux AArch64 FullAOT+LLVM

**Comment by lewing** [comment]
@monojenkins build failed

**Comment by lewing** [comment]
@monojenkins backport 3.2-wasm

**Pull request merged**